File: backend/src/main/java/bsuir/model/viewModel/TripRecord.java
package bsuir.model.viewModel;

import bsuir.model.stationDetails.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;


@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
public class TripRecord extends Trip{
    private Country departureCountryObj;
    private Country arrivalCountryObj;
    private City departureCityObj;
    private City arrivalCityObj;
    private Station departureStationObj;
    private Station arrivalStationObj;
    private TrainModel trainModelObj;
    private Train trainObj;

    public TripRecord(Trip trip){
        super(trip);
    }
}

File: frontend/src/app/models/view-models/trip-record.ts
import {Country} from "../station-details/country";
import {City} from "../station-details/city";
import {Station} from "../station-details/station";
import {Train} from "../station-details/train";
import {TrainModel} from "../station-details/train-model";
import {Trip} from "../station-details/trip";


export class TripRecord extends Trip {
  departureCountryObj: Country;
  arrivalCountryObj: Country;
  departureCityObj: City;
  arrivalCityObj: City;
  departureStationObj: Station;
  arrivalStationObj: Station;
  trainModelObj: TrainModel;
  trainObj: Train;
}
